core/arch/arm/kernel/mutex.c

```diff
@@ -102,6 +108,8 @@ static bool __mutex_trylock(struct mutex *m, const char *fname __unused,
 	uint32_t old_itr_status;
 	enum mutex_value old_value;
 
+	assert(thread_get_id_may_fail() != -1);
```

Comment:
I actually hesitated doing this. Technically, it seems that nothing would prevent `__mutex_trylock()` from succeeding even when a spinlock is held. But even so, would it make sense to allow such an operation? Considering that `__mutex_unlock()` is forbidden, I highly doubt it (locking hierarchy issues).
So, OK, I will add the assertion.